src/Features/Core/Portable/Navigation/INavigableItem.cs

```diff
@@ -45,5 +48,39 @@ internal interface INavigableItem
         bool IsStale { get; }
 
         ImmutableArray<INavigableItem> ChildItems { get; }
+
+        public record NavigableDocument(NavigableProject Project, string Name, string? FilePath, IReadOnlyList<string> Folders, DocumentId Id)
+        {
+            public required bool IsSourceGeneratedDocument { get; init; }
+            public required Workspace Workspace { get; init; }
+
+            public static NavigableDocument FromDocument(Document document)
+                => new(NavigableProject.FromProject(document.Project), document.Name, document.FilePath, document.Folders, document.Id)
+                {
+                    IsSourceGeneratedDocument = document is SourceGeneratedDocument,
+                    Workspace = document.Project.Solution.Workspace,
+                };
+
+            internal ValueTask<Document> GetDocumentAsync(Solution solution, CancellationToken cancellationToken)
+                => solution.GetRequiredDocumentAsync(Id, includeSourceGenerated: IsSourceGeneratedDocument, cancellationToken);
+
+            internal async ValueTask<SourceText> GetTextAsync(Solution solution, CancellationToken cancellationToken)
```

Comment:
GetRequiredText.

Reply:
➡️ `GetText` is always required in Roslyn.